xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp

```diff
@@ -27,20 +34,35 @@ std::unique_ptr<ISavestate> CSavestateDatabase::CreateSavestate()
   return savestate;
 }
 
-bool CSavestateDatabase::AddSavestate(const std::string& gamePath, const ISavestate& save)
+bool CSavestateDatabase::AddSavestate(std::string& savestatePath,
+                                      const std::string& gamePath,
+                                      const ISavestate& save)
 {
   bool bSuccess = false;
+  std::string path;
+
+  if (savestatePath.empty())
+  {
+    path = MakePath(gamePath);
+    path = URIUtils::AddFileToFolder(path, CDateTime::GetCurrentDateTime().GetAsSaveString() +
```

Comment:
(assumes apocalyptic tone) capturing system time is EVIL!!! I work in embedded, where system time can be wildly inaccurate with no internet. My company even takes the hit and includes an RTC on *all* of our devices to make system time less unwieldy.

To reflect this viewpoint, it's best to capture system time exactly once, and use that result everywhere. The `save` should include the time it was captured it, so let's use that instead of taking a new reading of the system clock.